presto-hive/src/main/java/com/facebook/presto/hive/parquet/predicate/ParquetPredicateUtils.java

```diff
@@ -188,14 +190,37 @@ private static boolean isColumnPredicate(ColumnDescriptor columnDescriptor, Tupl
                 .anyMatch(columnName::equals);
     }
 
-    private static boolean isOnlyDictionaryEncodingPages(Set<Encoding> encodings)
+    @VisibleForTesting
+    @SuppressWarnings("deprecation")
+    static boolean isOnlyDictionaryEncodingPages(Set<Encoding> encodings)
     {
-        // more than 1 encodings for values
-        if (encodings.size() > PARQUET_DATA_TRIPLE) {
+        if (encodings.contains(PLAIN_DICTIONARY)) {
+            // if remove returned true, PLAIN_DICTIONARY was present, which means at
```

Comment:
This was adapted from another patch that used remove instead of contains. I've updated the comment.